config/initializers/active_model_serializer.rb

```diff
@@ -1,3 +1,46 @@
 # frozen_string_literal: true
 
-ActiveModelSerializers.config.adapter = :json_api
+class CustomPaginationLinks < ActiveModelSerializers::Adapter::JsonApi
+  def success_document
+    res = super
+
+    unless res[:links].nil?
+      normalize_pagination_links(res[:links])
+
+      # Only append pagination metadata if no data has been set yet
+      if res[:meta].nil?
+        res[:meta] = append_pagination_meta_info(@serializer.object)
+      end
+    end
+
+    return res
+  end
+
+  private
+
+  def append_pagination_meta_info(collection)
+    {
+      'total-pages': collection.total_pages,
+      'total-items': collection.total_entries,
+      size: collection.per_page,
+    }
+  end
+
+  def normalize_pagination_links(links)
+    links[:self] = replace_dashboard_correct_url(replace_logged_user_query_param(links[:self]))
+    links[:first] = replace_dashboard_correct_url(replace_logged_user_query_param(links[:first]))
+    links[:last] = replace_dashboard_correct_url(replace_logged_user_query_param(links[:last]))
+    links[:prev] = replace_dashboard_correct_url(replace_logged_user_query_param(links[:prev]))
+    links[:next] = replace_dashboard_correct_url(replace_logged_user_query_param(links[:next]))
+  end
+
+  def replace_logged_user_query_param(str)
+    return !str.nil? ? str.gsub(/loggedUser=[^&]*&/m, '') : str
```

Comment:
Rule of thumb: double negatives ( ! .nil? ) are evil, because the human brain is weak. Until we all get converted into cyborgs, you should opt for something like 

´´´
return str if str.nil?
str.gsub(/loggedUser=[^&]*&/m, '')
´´´